app/src/androidTest/kotlin/at/bitfire/davdroid/sync/SyncerTest.kt

```diff
@@ -66,19 +52,43 @@ class SyncerTest {
     }
 
 
-    class TestSyncer @Inject constructor() : Syncer() {
+    inner class TestSyncer : Syncer<LocalTestCollection>(account, arrayOf(), SyncResult()) {
 
         val syncCalled = AtomicInteger()
+        override val serviceType: String
+            get() = TODO("Not yet implemented")
+        override val authority: String
+            get() = TODO("Not yet implemented")
+        override val localCollections: List<LocalTestCollection>
+            get() = TODO("Not yet implemented")
+        override val localSyncCollections: List<LocalTestCollection>
+            get() = TODO("Not yet implemented")
+
+        override fun beforeSync() {
+            syncCalled.incrementAndGet()
+        }
 
-        override fun sync(
-            account: Account,
-            extras: Array<String>,
-            authority: String,
-            httpClient: Lazy<HttpClient>,
-            provider: ContentProviderClient,
-            syncResult: SyncResult
+        override fun create(remoteCollection: Collection) {
+            TODO("Not yet implemented")
+        }
+
+        override fun syncCollection(
+            localCollection: LocalTestCollection,
+            remoteCollection: Collection
         ) {
-            syncCalled.incrementAndGet()
+            TODO("Not yet implemented")
+        }
+
+        override fun update(localCollection: LocalTestCollection, remoteCollection: Collection) {
+            TODO("Not yet implemented")
+        }
+
+        override fun delete(localCollection: LocalTestCollection) {
+            TODO("Not yet implemented")
+        }
+
+        override fun getUrl(localCollection: LocalTestCollection): HttpUrl? {
+            TODO("Not yet implemented")
         }
```

Comment:
I wonder whether it would be possible to use mockk instead?

Reply:
Maybe, I will try :)